### DISCUSSION TITLE
Trouble with Micropython & FreeRTOS on STM32

### DISCUSSION THREAD

**Discussion by Ger-zal** [micropython/micropython#12467] [Core Development]
Did anyone succeed in implementing micropython as a task (freertos) ? Because I cant figure it out on my stm32l4x6 ( I saw the esp32 and cc3200).

I Saw many questions on différent forum but no real exemple. I'm trying a basic micropython as task + led blink task.

Can someone share a minimalistic code or changes for the stm32 port ?

**Comment by jimmo** [comment]
@Ger-zal converted this to a discussion. Just for my own curiosity, did GitHub show the new issue template, and the placeholder text for the feature request template?

> ( I saw the esp32 and cc3200).

On the esp32 and cc3200, the vendor SDK is built around FreeRTOS so this is essentially the only way to run MicroPython on them. On STM32 we just run directly (essentially MicroPython is "the OS").

> I Saw many questions on différent forum but no real exemple. I'm trying a basic micropython as task + led blink task.
> Can someone share a minimalistic code

This is what we're trying to address with the "embed" port -- i.e. to embed MicroPython in an existing firmware (and build system).

https://github.com/micropython/micropython/tree/master/ports/embed
https://github.com/micropython/micropython/tree/master/examples/embedding

And also worth looking at https://github.com/micropython/micropython/pull/11430

This should be enough to run MicroPython in a task, and then you just need to provide bindings to whatever else your firmware does (e.g. access to LEDs or whatever).

> or changes for the stm32 port ?

It will be very difficult to adapt the STM32 port to the scenario you're describing -- it's really intended to run bare-metal.

**Comment by bidrohini** [comment]
Here are only a few info: https://forum.micropython.org/viewtopic.php?t=12692
However, you can consider posting this question to the STM forum too.
